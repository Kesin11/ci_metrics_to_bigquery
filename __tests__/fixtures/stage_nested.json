{
  "_links": {
    "self": {
      "href": "/job/build-pipeline/28/wfapi/describe"
    }
  },
  "id": "28",
  "name": "#28",
  "status": "SUCCESS",
  "startTimeMillis": 1581160762770,
  "endTimeMillis": 1581161083628,
  "durationMillis": 320858,
  "queueDurationMillis": 2,
  "pauseDurationMillis": 0,
  "stages": [
    {
      "_links": {
        "self": {
          "href": "/job/build-pipeline/28/execution/node/6/wfapi/describe"
        }
      },
      "id": "6",
      "name": "Show env",
      "execNode": "",
      "status": "SUCCESS",
      "startTimeMillis": 1581161074050,
      "durationMillis": 302,
      "pauseDurationMillis": 0,
      "stageFlowNodes": [
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/7/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/7/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/7/log"
            }
          },
          "id": "7",
          "name": "Shell Script",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "env",
          "startTimeMillis": 1581161074075,
          "durationMillis": 266,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "6"
          ]
        }
      ]
    },
    {
      "_links": {
        "self": {
          "href": "/job/build-pipeline/28/execution/node/11/wfapi/describe"
        }
      },
      "id": "11",
      "name": "Setup",
      "execNode": "",
      "status": "SUCCESS",
      "startTimeMillis": 1581161074361,
      "durationMillis": 3056,
      "pauseDurationMillis": 0,
      "stageFlowNodes": [
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/14/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/14/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/14/log"
            }
          },
          "id": "14",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "3.73807278690767",
          "startTimeMillis": 1581161074388,
          "durationMillis": 3007,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "13"
          ]
        }
      ]
    },
    {
      "_links": {
        "self": {
          "href": "/job/build-pipeline/28/execution/node/20/wfapi/describe"
        }
      },
      "id": "20",
      "name": "build",
      "execNode": "",
      "status": "SUCCESS",
      "startTimeMillis": 1581161077424,
      "durationMillis": 4065,
      "pauseDurationMillis": 0,
      "stageFlowNodes": [
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/23/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/23/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/23/log"
            }
          },
          "id": "23",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "Random t = 0.5388634603140408",
          "startTimeMillis": 1581161077449,
          "durationMillis": 5,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "22"
          ]
        },
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/24/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/24/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/24/log"
            }
          },
          "id": "24",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "startTimeMillis": 1581161077454,
          "durationMillis": 3,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "23"
          ]
        },
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/25/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/25/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/25/log"
            }
          },
          "id": "25",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "4.53886346031404",
          "startTimeMillis": 1581161077457,
          "durationMillis": 4008,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "24"
          ]
        }
      ]
    },
    {
      "_links": {
        "self": {
          "href": "/job/build-pipeline/28/execution/node/31/wfapi/describe"
        }
      },
      "id": "31",
      "name": "Declarative: Post Actions",
      "execNode": "",
      "status": "SUCCESS",
      "startTimeMillis": 1581161081498,
      "durationMillis": 2034,
      "pauseDurationMillis": 0,
      "stageFlowNodes": [
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/32/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/32/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/32/log"
            }
          },
          "id": "32",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "success",
          "startTimeMillis": 1581161081510,
          "durationMillis": 4,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "31"
          ]
        },
        {
          "_links": {
            "self": {
              "href": "/job/build-pipeline/28/execution/node/33/wfapi/describe"
            },
            "log": {
              "href": "/job/build-pipeline/28/execution/node/33/wfapi/log"
            },
            "console": {
              "href": "/job/build-pipeline/28/execution/node/33/log"
            }
          },
          "id": "33",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "2",
          "startTimeMillis": 1581161081514,
          "durationMillis": 2009,
          "pauseDurationMillis": 0,
          "parentNodes": [
            "32"
          ]
        }
      ]
    }
  ]
}
