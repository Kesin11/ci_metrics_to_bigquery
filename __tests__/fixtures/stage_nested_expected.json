{
  "id": "28",
  "name": "#28",
  "status": "SUCCESS",
  "startTime": "2020-02-08T11:19:22Z",
  "endTime": "2020-02-08T11:24:43Z",
  "startTimeMillis": 1581160762770,
  "endTimeMillis": 1581161083628,
  "durationMillis": 320858,
  "duration": 320.858,
  "queueDurationMillis": 2,
  "queueDuration": 0.002,
  "pauseDurationMillis": 0,
  "pauseDuration": 0,
  "stages": [
    {
      "id": "6",
      "name": "Show env",
      "execNode": "",
      "status": "SUCCESS",
      "startTime": "2020-02-08T11:24:34Z",
      "startTimeMillis": 1581161074050,
      "durationMillis": 302,
      "duration": 0.302,
      "pauseDurationMillis": 0,
      "pauseDuration": 0,
      "stageFlowNodes": [
        {
          "id": "7",
          "name": "Shell Script",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "env",
          "startTime": "2020-02-08T11:24:34Z",
          "startTimeMillis": 1581161074075,
          "durationMillis": 266,
          "duration": 0.266,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "6"
          ]
        }
      ]
    },
    {
      "id": "11",
      "name": "Setup",
      "execNode": "",
      "status": "SUCCESS",
      "startTime": "2020-02-08T11:24:34Z",
      "startTimeMillis": 1581161074361,
      "durationMillis": 3056,
      "duration": 3.056,
      "pauseDurationMillis": 0,
      "pauseDuration": 0,
      "stageFlowNodes": [
        {
          "id": "14",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "3.73807278690767",
          "startTime": "2020-02-08T11:24:34Z",
          "startTimeMillis": 1581161074388,
          "durationMillis": 3007,
          "duration": 3.007,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "13"
          ]
        }
      ]
    },
    {
      "id": "20",
      "name": "build",
      "execNode": "",
      "status": "SUCCESS",
      "startTime": "2020-02-08T11:24:37Z",
      "startTimeMillis": 1581161077424,
      "durationMillis": 4065,
      "duration": 4.065,
      "pauseDurationMillis": 0,
      "pauseDuration": 0,
      "stageFlowNodes": [
        {
          "id": "23",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "Random t = 0.5388634603140408",
          "startTime": "2020-02-08T11:24:37Z",
          "startTimeMillis": 1581161077449,
          "durationMillis": 5,
          "duration": 0.005,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "22"
          ]
        },
        {
          "id": "24",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "startTime": "2020-02-08T11:24:37Z",
          "startTimeMillis": 1581161077454,
          "durationMillis": 3,
          "duration": 0.003,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "23"
          ]
        },
        {
          "id": "25",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "4.53886346031404",
          "startTime": "2020-02-08T11:24:37Z",
          "startTimeMillis": 1581161077457,
          "durationMillis": 4008,
          "duration": 4.008,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "24"
          ]
        }
      ]
    },
    {
      "id": "31",
      "name": "Declarative: Post Actions",
      "execNode": "",
      "status": "SUCCESS",
      "startTime": "2020-02-08T11:24:41Z",
      "startTimeMillis": 1581161081498,
      "durationMillis": 2034,
      "duration": 2.034,
      "pauseDurationMillis": 0,
      "pauseDuration": 0,
      "stageFlowNodes": [
        {
          "id": "32",
          "name": "Print Message",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "success",
          "startTime": "2020-02-08T11:24:41Z",
          "startTimeMillis": 1581161081510,
          "durationMillis": 4,
          "duration": 0.004,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "31"
          ]
        },
        {
          "id": "33",
          "name": "Sleep",
          "execNode": "",
          "status": "SUCCESS",
          "parameterDescription": "2",
          "startTime": "2020-02-08T11:24:41Z",
          "startTimeMillis": 1581161081514,
          "durationMillis": 2009,
          "duration": 2.009,
          "pauseDurationMillis": 0,
          "pauseDuration": 0,
          "parentNodes": [
            "32"
          ]
        }
      ]
    }
  ]
}
